src/ethereum_test_specs/eof.py

```diff
@@ -237,22 +238,43 @@ def verify_result(self, result: CompletedProcess, expected_result: Result, code:
                 raise UnexpectedEOFException(
                     code=code, got=f"{actual_exception} ({actual_message})"
                 )
+        elif isinstance(expected_result.exception, List):
+            expected_string = to_pipe_str(expected_result.exception)
+            if "OK" in actual_message:
+                raise ExpectedEOFException(
+                    code=code,
+                    expected=f"multiple possible exceptions {expected_string}",
+                    got="valid EOF code",
+                )
 
-        expected_exception = expected_result.exception
-        expected_message = parser.exception_to_message(expected_exception)
-
-        if "OK" in actual_message:
-            raise ExpectedEOFException(
-                code=code, expected=f"{expected_exception} ({expected_message})"
-            )
+            for expected in expected_result.exception:
+                if expected == actual_exception:
+                    return
```

Comment:
```suggestion
            if actual_exception in expected_result.exception:
                return
```
This should do the trick, and it also works for non-list exceptions.

Reply:
Done, and dropped it back to one case for single or multiple exceptions